src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerGroupRecipe.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.composer.internal
+
+import javax.annotation.Nonnull
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Provider
+import javax.inject.Singleton
+
+import org.sonatype.nexus.repository.Format
+import org.sonatype.nexus.repository.Repository
+import org.sonatype.nexus.repository.Type
+import org.sonatype.nexus.repository.group.GroupFacetImpl
+import org.sonatype.nexus.repository.group.GroupHandler
+import org.sonatype.nexus.repository.http.HttpHandlers
+import org.sonatype.nexus.repository.types.GroupType
+import org.sonatype.nexus.repository.view.ConfigurableViewFacet
+import org.sonatype.nexus.repository.view.Router
+import org.sonatype.nexus.repository.view.ViewFacet
+
+/**
+ * Recipe for creating a Composer group repository.
+ */
+@Named(ComposerGroupRecipe.NAME)
+@Singleton
+class ComposerGroupRecipe
+    extends ComposerRecipeSupport
+{
+  public static final String NAME = 'composer-group'
+
+  @Inject
+  Provider<GroupFacetImpl> groupFacet
+
+  @Inject
+  GroupHandler standardGroupHandler
+
+  @Inject
+  ComposerGroupPackagesJsonHandler packagesJsonHandler
+
+  @Inject
+  ComposerGroupProviderJsonHandler providerJsonHandler
+
+  @Inject
+  ComposerGroupRecipe(@Named(GroupType.NAME) final Type type, @Named(ComposerFormat.NAME) final Format format) {
+    super(type, format)
+  }
+
+  @Override
+  void apply(@Nonnull final Repository repository) throws Exception {
+    repository.attach(groupFacet.get())
+    repository.attach(storageFacet.get())
+    repository.attach(securityFacet.get())
+    repository.attach(configure(viewFacet.get()))
+    repository.attach(attributesFacet.get())
+  }
+
+  /**
+   * Configure {@link ViewFacet}.
+   */
+  private ViewFacet configure(final ConfigurableViewFacet facet) {
+    Router.Builder builder = new Router.Builder()
+
+    builder.route(packagesMatcher()
+        .handler(timingHandler)
+        .handler(assetKindHandler.rcurry(AssetKind.PACKAGES))
```

Comment:
Trivial: no reason not to statically import these asset kinds.

Reply:
> Trivial: no reason not to statically import these asset kinds.

I left them without static imports in the other recipes; to me it does provide a bit of useful context, but I'm not averse to changing them as part of a tidying PR at some point.